hack/update-submodule.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Update submodules HEAD to remote branches
+
+set -euo pipefail
+
+function help() {
+    echo "Usage: update_submodules.sh [OPTIONS]... branch_name..."
+    echo "Update submodules HEAD to remote branches"
+    echo "Must be ran in repo root directory"
+    echo ""
+    echo "Options:"
+    echo "-h   Shows usage text"
+    echo "Example:"
+    echo "update_submodules.sh master release-4.6"
+}
+
+function update_branch_submodules() {
+    branch=$1
+    working_branch_name="$branch-SubmoduleUpdate$(date +%m-%d)"
+    echo Using branch $working_branch_name
+    git branch -D $working_branch_name || true
+    git checkout $branch -b $working_branch_name
+    git submodule update --remote
+
+    # The submodules have been updated at this point, now a commit needs to be generated for each of them
```

Comment:
Should we trying to update all of them at the same time? How is this going to work with `windows_exporter`?

Reply:
I dont see a problem with updating them all at the same time.

I'm not sure what the issue is with `windows_exporter`? Each submodule should have its proper remote set up in the config.

Reply:
`windows_exporter` only has a `master` branch today. So what does the script do today?

In general I feel there should updating one submodule or all submodules should be an option. Take the instance we just want to update one submodule because of a bug fix and don't want the other submodules updated.